packages/core/src/components/TextArea/TextArea.types.ts

```diff
@@ -7,7 +7,23 @@ type TextAreaNativeInputProps = Omit<
   "role" | "aria-describedby" | "aria-invalid"
 >;
 
-export interface TextAreaProps extends TextAreaNativeInputProps, VibeComponentProps {
+/**
+ * This type ensures that allowExceedingMaxLength is only used when maxLength is provided.
+ */
+type MaxLengthProps =
```

Comment:
As discussed, let's revert and use jsdoc comment to make the relation between these prop clearer